lib/datadog/profiling/component.rb

```diff
@@ -191,6 +189,47 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
 
         true
       end
+
+      # Versions of libmysqlclient prior to 8.0.0 are known to have buggy handling of system call interruptions.
+      # The profiler can sometimes cause system call interruptions, and so this combination can cause queries to fail.
+      #
+      # See https://bugs.mysql.com/bug.php?id=83109 and
+      # https://docs.datadoghq.com/profiler/profiler_troubleshooting/ruby/#unexpected-run-time-failures-and-errors-from-ruby-gems-that-use-native-extensions-in-dd-trace-rb-1110
+      # for details.
+      #
+      # The `mysql2` gem's `info` method can be used to determine which `libmysqlclient` version is in use, and thus to
+      # detect if it's safe for the profiler to use signals or if we need to employ a fallback.
+      private_class_method def self.compatible_libmysqlclient_version?(settings)
```

Comment:
We could rename the method to `incompatible_libmysqlclient_version?` that way in the check above we do not have to negate the result

Reply:
Changed in c44e9a7991a32fb1e5a0e264047999ffffd58ab5